Theme import and update to newer hugo, content reorg

abrokenjester opened the pull request:
Fixes eclipse/rdf4j#1902
Fixes eclipse/rdf4j#2201

- hugo solstice theme checked in as git submodule to facilitate easier future updates
- moved all previous customizations we did in the the theme itself to our own project overrides
- making better use of Hugo taxonomies, sections, and list features to organize content:
    - each overview page is now a list (denoted by having an `_index.md` frontmatter page). 
    - rendering defaults for documentation lists and release note lists are added to the layouts dir
- replaced all uses of raw HTML with corresponding markdown to make version bump to latest Hugo possible
- replaced `{{ < highlight java > }}` with markdown equivalent. 
- gave documentation overview page a slightly nicer look (YMMV)
- release notes now organized as one page per release, allowing for easier handling of larger release notes in future. 

I have not yet tested this on our actual web server, but running hugo locally produces a site that looks how I expect it to look. Nice bonus is that the new markdown engine in Hugo apparently even recognizes sparql and turtle, and does syntax highlighting for it. If you have time, try it out locally: just run `hugo server` from the site/ directory and you'll be able to see the site on localhost:1313.

abrokenjester committed:
GH-2201 remove hard checkin of copy of solstice theme

abrokenjester committed:
eclipse/rdf4j#2201 theme as submodule

- includes necessary changes to content and styles
- prep for eclipse/rdf4j#1902 hugo version bump

abrokenjester committed:
eclipse/rdf4j#1902 replace raw html with markdown

abrokenjester committed:
GH-2201 use https url for submodule

jeenbroekstra commented:
I've currently made this branch live, so you can see what it looks like if you go to https://rdf4j.org/ . 

Not normally how we do business of course but I had some issues getting new hugo installed on the server and getting it to compile the site, so I needed to do a live test. Might as well keep it up for now. 

The site's auto-update is currently disabled, as soon as we merge this I'll tweak it to re-enable automatic updates.

abrokenjester committed:
GH-1902 re-enable raw html rendering

- there are some things just too hard to do with plain markdown
- we should minimize our reliance on raw html where possible

abrokenjester committed:
GH-2201 disable js param to fix hamburger menu on mobile

hmottestad commented:
Looks good!

Btw. do you know how to change the overflow from wrap to scroll on the code blocks. This was like this from before, your branch doesn’t change anything. 

![B17C49AA-8E0D-4CF6-9418-773753EB9605](https://user-images.githubusercontent.com/797185/81467951-77fb1e00-91dc-11ea-8bd2-5b57cd355032.jpeg)

jeenbroekstra commented:
> Looks good!
> 
> Btw. do you know how to change the overflow from wrap to scroll on the code blocks. This was like this from before, your branch doesn’t change anything.
> 

I'm not sure. This is perhaps something we can tweak in the CSS but I'd have to dive in to figure it out.
